### ISSUE TITLE
Size models and finetuned networks

### ISSUE THREAD

**Issue by Cactusfacts** [MouseLand/cellpose#632]
Hello,

I've started investigating the size models, starting with training a size model for a finetuned Cellpose model (based on cyto2 weights). I notice that the size model is using my finetuned model's diam_mean, when I would expect it to use diam_labels:

https://github.com/MouseLand/cellpose/blob/4e8205125750c0c82e03386f28ff6d4bef1da6c7/cellpose/models.py#L820

the initial cyto2 model has diam_mean = diam_labels = 30. My finetuned model has diam_mean = 30, diam_labels=46. Shouldn't we use  cp.diam_labels to train the size model instead, since the CellposeModel's eval() routine uses this to calculate its resize factor? Also, does the finetuned model really need to know about the original diam_mean of 30 anymore?

I also noticed some issues with the __main__.py script's --train_size code not being compatible with the newest version of the size model code (__main__.py does some label manipulation that SizeModel.train isn't expecting), so I was wondering if size models are still considered meaningful and useful in the new cellpose 2.0 finetuning paradigm, or if they're more deprecated at this point. Thanks in advance for your thoughts!

**Comment by carsen-stringer** [comment]
Good questions, TLDR we don't see the size models as useful anymore as users will train their own models and the diameter will default to the average diameter of their training set.

the diam_mean is even more internal, you would have to change that when the model is created. Cellpose always resizes images to the diam_mean of 30 pixels (or a different diam_mean if you set it). during training, this means that Cellpose takes the mean of the diameter of objects in the image to determine the rescale size and rescales accordingly. So Cellpose is trained on resized versions of your images, not images with objects of size 46.

Now with the trained model, your diam_labels is used to determine the rescale factor each time the Cellpose network runs.

**Comment by Cactusfacts** [comment]
Much appreciated Carsen! 

I think what I missed is that step 1 of the size model's eval() extracts the feature vectors from the nn with a resize based on diam_labels. 

I take your point about the finetuning workflow obviating the need for size models in relatively homogeneous training sets. Do you think a size model-type approach could still have utility in highly heterogeneous assays with larger size & phenotypic variations? Or could we abstract the approach beyond just diameters and develop a style vector(s)->best finetuned CP model mapping to pick the right specialist cellpose model per image?
